File: src/main/java/co/uk/jiveelection/campaign/jive/ge2015/MilibandJive.java
package co.uk.jiveelection.campaign.jive.ge2015;

import co.uk.jiveelection.campaign.TwitConfig;
import co.uk.jiveelection.campaign.input.twitter.TwitterInput;
import co.uk.jiveelection.campaign.jive.Jive;
import co.uk.jiveelection.campaign.output.twitter.TwitterOutput;
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslator;
import twitter4j.*;
import twitter4j.conf.Configuration;
import twitter4j.conf.ConfigurationBuilder;

public class MilibandJive extends Jive<TwitterInput, TwitterOutput, InMemoryJiveTranslator> {

    public MilibandJive() {
        final Configuration configuration = new ConfigurationBuilder()
                .setOAuthConsumerKey(TwitConfig.CONSUMER_TOKEN)
                .setOAuthConsumerSecret(TwitConfig.CONSUMER_TOKEN_SECRET)
                .setOAuthAccessToken(TwitConfig.ED_JIVEABAND_ACCESS_TOKEN)
                .setOAuthAccessTokenSecret(TwitConfig.ED_JIVEABAND_ACCESS_TOKEN_SECRET)
                .build();

        final Twitter jiveMilibandTwitter = new TwitterFactory(configuration).getInstance();
        final TwitterStream jiveMilibandTwitterStream = new TwitterStreamFactory(configuration).getInstance();

        input = new TwitterInput(this, "Ed_Miliband", jiveMilibandTwitter, jiveMilibandTwitterStream);
        output = new TwitterOutput(this, jiveMilibandTwitter);
        translator = new InMemoryJiveTranslator(this);
    }

    @Override
    public void init() throws TwitterException {
        input.init();
    }

    @Override
    public void onJiveComplete(Object o) {
        // Do nothing for now.
    }
}

File: src/main/kotlin/co/uk/jiveelection/campaign/jive/ge2015/MilibandJive.kt
package co.uk.jiveelection.campaign.jive.ge2015

import co.uk.jiveelection.campaign.input.twitter.TwitterInput
import co.uk.jiveelection.campaign.input.twitter.TwitterInputFactory
import co.uk.jiveelection.campaign.jive.Jive
import co.uk.jiveelection.campaign.output.twitter.TwitterOutput
import co.uk.jiveelection.campaign.output.twitter.TwitterOutputFactory
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslator
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslatorFactory
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import twitter4j.TwitterException

import javax.inject.Inject

class MilibandJive @Inject
constructor(
        twitterInputFactory: TwitterInputFactory,
        twitterOutputFactory: TwitterOutputFactory,
        inMemoryJiveTranslatorFactory: InMemoryJiveTranslatorFactory
) : Jive<TwitterInput, TwitterOutput, InMemoryJiveTranslator>() {

    private val logger: Logger = LoggerFactory.getLogger(MilibandJive::class.java)

    override val input: TwitterInput = twitterInputFactory.create(this)
    override val output: TwitterOutput = twitterOutputFactory.create(this)
    override val translator: InMemoryJiveTranslator = inMemoryJiveTranslatorFactory.create(this)


    @Throws(TwitterException::class)
    override fun init() {
        input.init()
        logger.info("MilibandJive initialised.")
    }

    override fun onJiveComplete(o: Any) {
        // Do nothing for now.
    }
}
